pinot-core/src/main/java/com/linkedin/pinot/core/data/manager/offline/AbstractTableDataManager.java

```diff
@@ -260,4 +263,16 @@ public void releaseSegment(SegmentDataManager segmentDataManager) {
     }
   }
 
+  @Override
+  public @Nonnull Set<String> getSegmentNameList() {
+    if (_segmentsMap == null) {
+      return new HashSet<>();
+    }
+    return _segmentsMap.keySet();
```

Comment:
Reading keySet() API, says that iterating over the set entries can result in undefined behavior if the hash underneath is changed.

We modify the _segmentsMap under a read-write lock, so that we can atomically add/remove segments while keeping ref cnt.

If this is only used for debug, perhaps it is best to get a readlock, create a new set with the entries, and release the original map? Most of the time we are getting readlock (for queries), so it should be ok to get a readlock once in a while for debug commands.

Segment add/remove takes write lock